cni/internal/cniutil.go

```diff
@@ -61,3 +64,41 @@ func IfacesWithName(name string, ifaces ...*current.Interface) []*current.Interf
 
 	return foundIfaces
 }
+
+// VMTapPair takes a CNI result and returns the tap iface and vm iface corresponding
+// to the provided vmID. See the vmconf package docs for details on the expected
+// vm and tap iface configurations.
+func VMTapPair(
```

Comment:
This function declaration looks super weird.  Would this look better?

```go
func VMTapPair(
	result *current.Result,
	vmID string,
)(
	vmIface *current.Interface,
	tapIface *current.Interface,
	err error
)
```

Also: can we avoid the named returns?

Reply:
Sure I don't have a preference either way for both.

Reply:
Updated the format of the declaration, but as I was editing I remembered I used the named returns because I think it makes it clear which arg is which since the first two are both of the same type.

Given that, I think this is an okay use of named returns because there are no naked return statements (all of the args are always fully specified in each `return`) and there's no weirdness happening with defers modifying the return values or something like that. Let me know if you disagree.

Reply:
I tend to dislike named returns and use them only when necessary (defers modifying the returns is one use-case).  You did avoid naked returns; naked returns make it much harder to reason about what is actually returned.  But the use of named returns make it possible for future introduction of a naked return by someone else.

I don't feel strongly enough here to want to force you to change it, but I avoid named returns as much as possible.